src/lib/components/toggle/QSwitch.svelte

```diff
@@ -0,0 +1,110 @@
+<script lang="ts">
+  import QIconSnippet from "$components/private/QIconSnippet.svelte";
+  import { ripple } from "$helpers";
+  import { isActivationKey } from "$utils";
+  import type { QSwitchProps } from "./props";
+
+  let {
+    value = $bindable(),
+    label = undefined,
+    labelPosition = "right",
+    disabled = false,
+    icons = false,
+    showOnlyCheckedIcon = false,
+    checkedIcon,
+    uncheckedIcon,
+    ...props
+  }: QSwitchProps = $props();
+
+  let qSwitch: HTMLDivElement;
+  let qSwitchInput: HTMLInputElement;
+
+  function toggle() {
+    value = !value;
+  }
+
+  function onclick() {
```

Comment:
Probably better to name it `onClick` and also check for `onclick` from props?

Reply:
7323a5c3a8386538513c5a3fa80d5ca4a7fa0273